roles/platform/tasks/setup_azure_authz.yml

```diff
@@ -108,128 +108,103 @@
     role_definition_id: "{{ __azure_rl_assgn_item.role }}"
   loop:
-    - role: "{{ __azure_virtualmachine_ctrb_role_uri }}"
-      name: "{{ plat__azure_vmcontributor_idbroker_assn_name }}"
+    - role: "{{ __azure_virtualmachine_ctrb_role_id }}"
       scope: "{{ plat__azure_subscription_uri }}"
-      assignee: "{{ __azure_idbroker_identity_uuid }}"
+      assignee: "{{ __azure_idbroker_identity.properties.principalId }}"
       desc: Assign VM Contributor Role to IDBroker Identity at Subscription Level
-    - role: "{{ __azure_managedidentity_optr_role_uri }}"
-      name: "{{ plat__azure_mgdidentop_idbroker_assn_name }}"
+    - role: "{{ __azure_managedidentity_optr_role_id }}"
       scope: "{{ plat__azure_subscription_uri }}"
-      assignee: "{{ __azure_idbroker_identity_uuid }}"
+      assignee: "{{ __azure_idbroker_identity.properties.principalId }}"
       desc: Assign Managed Identity Operator Role to IDBroker Identity at Subscription Level
-    - role: "{{ __azure_storageblobdata_ownr_role_uri }}"
-      name: "{{ plat__azure_storageowner_datalakeadmin_logs_assn_name }}"
+    - role: "{{ __azure_storageblobdata_ownr_role_id }}"
       scope: "{{ plat__azure_logpath_uri }}"
-      assignee: "{{ __azure_datalakeadmin_identity_uuid }}"
+      assignee: "{{ __azure_datalakeadmin_identity.properties.principalId }}"
       desc: Assign Storage Blob Data Owner Role to Data Lake Admin Identity at Logs Container Level
-    - role: "{{ __azure_storageblobdata_ownr_role_uri }}"
-      name: "{{ plat__azure_storageowner_datalakeadmin_data_assn_name }}"
+    - role: "{{ __azure_storageblobdata_ownr_role_id }}"
       scope: "{{ plat__azure_datapath_uri }}"
-      assignee: "{{ __azure_datalakeadmin_identity_uuid }}"
+      assignee: "{{ __azure_datalakeadmin_identity.properties.principalId }}"
       desc: Assign Storage Blob Data Owner Role to Data Lake Admin Identity at Data Container Level
-    - role: "{{ __azure_storageblobdata_ctrb_role_uri }}"
-      name: "{{ plat__azure_storagecontr_log_assn_name }}"
+    - role: "{{ __azure_storageblobdata_ctrb_role_id }}"
       scope: "{{ plat__azure_logpath_uri }}"
-      assignee: "{{ __azure_log_identity_uuid  }}"
+      assignee: "{{ __azure_log_identity.properties.principalId }}"
       desc: Assign Storage Blob Data Contributor Role to Log Role at Logs Container level
-    - role: "{{ __azure_storageblobdata_ctrb_role_uri }}"
-      name: "{{ plat__azure_storagecontr_ranger_audit_assn_name }}"
+    - role: "{{ __azure_storageblobdata_ctrb_role_id }}"
       scope: "{{ plat__azure_datapath_uri }}"
-      assignee: "{{ __azure_ranger_audit_identity_uuid }}"
-      desc: Assign Storage Blob Data Contributor Role to Ranger Role at Data Container level
+      assignee: "{{ __azure_ranger_audit_identity.properties.principalId }}"
+      desc: Assign Storage Blob Data Contributor Role to Ranger Audit Role at Data Container level
+    - role: "{{ __azure_contributor_role_id }}"
```

Comment:
lines 204-207 should be removed. It's a potential double-assignment of the cross account app.